src/Behat/Context/CookieComplianceContext.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Metadrop\Behat\Context;
+
+use Behat\MinkExtension\Context\RawMinkContext;
+use Behat\Mink\Element\NodeElement;
+
+/**
+ * Allows checking that the sites are cookie gdpr compliant.
+ *
+ * It checks that before accepting cookies there are not cookies
+ * saved in the browser, and when the cookies are accepted, the
+ * expected cookies appears.
+ */
+class CookieComplianceContext extends RawMinkContext {
+
+  /**
+   * Selector that accepts default cookie categories.
+   *
+   * @var string
+   */
+  protected string $cookieAgreeSelector;
+
+  /**
+   * Cookie banner selector.
+   *
+   * @var string
+   */
+  protected string $cookieBannerSelector;
+
+  /**
+   * List of cookies classified with category.
+   *
+   * @var array
+   */
+  protected array $cookies;
+
+  /**
+   * Cookie compliance constructor.
+   *
+   * @param string $cookie_agree_selector
+   *   Selector that accepts default cookie categories.
+   * @param string $cookie_banner_selector
+   *   Cookie banner selector.
+   * @param array $cookies
+   *   List of cookies classified with category.
+   */
+  public function __construct(string $cookie_agree_selector, string $cookie_banner_selector, array $cookies) {
+    $this->cookieAgreeSelector = $cookie_agree_selector;
+    $this->cookieBannerSelector = $cookie_banner_selector;
+    $this->cookies = $cookies;
+  }
+
+  /**
+   * Accept cookie compliance.
+   *
+   * It accepts it by accepting all the categories
+   * that are accepted by default.
+   *
+   * @Then I accept cookies
+   */
+  public function iAcceptCookies() {
+    $agree_button = $this->getSession()->getPage()->find('css', $this->cookieAgreeSelector);
+    if ($agree_button instanceof NodeElement) {
+      $agree_button->press();
+      if (!$this->getSession()->wait(10000, sprintf('document.querySelector("%s") == null', $this->cookieBannerSelector))) {
+        throw new \Exception(sprintf('The cookie banner with selector "%s" is stil present after accepting cookies.', $this->cookieBannerSelector));
+      }
+    }
+    else {
+      throw new \Exception('The agree button do not appears.');
+    }
+  }
+
+  /**
+   * Check the main cookies of a specific type are not present.
+   *
+   * @Then there should not be cookies of :type type
```

Comment:
Does it make sense to support also the step "There should not be any cookie" ?
Not sure, but maybe it is helpful on the first site page load as no cookies should be present at all, I think that the mandatory ones are loaded once the user starts interacting. (Of course there will be exceptions)

Reply:
I think it can be useful in certain circumstances, there may be situations where no cookie is necessary, and the behavior of not loading any is desirable.

Reply:
After checking, the behat mink driver interface do not support getting all cookies. But, the web driver session that selenium web driver instantiates has a method to get all cookies. So I will check if it works fine, and if so, create a step to check there are no cookies that will only work with the Selenium web driver.

Reply:
Step added with a fully report of cookies present and potential cookies source like iframes